src/server/hotspot_partition_calculator.cpp

```diff
@@ -98,5 +105,52 @@ void hotspot_partition_calculator::data_analyse()
     }
 }
 
+// TODO:（TangYanzhao) call this function to start hotkey detection
+/*static*/ void hotspot_partition_calculator::send_hotkey_detect_request(
+    const std::string &app_name,
+    const uint64_t partition_index,
+    const dsn::apps::hotkey_type::type hotkey_type,
+    const dsn::apps::hotkey_detect_action::type action)
+{
+    auto request = std::make_unique<dsn::apps::hotkey_detect_request>();
+    request->type = hotkey_type;
+    request->action = action;
+    ddebug_f("{} {} hotkey detection in {}.{}",
+             (action == dsn::apps::hotkey_detect_action::STOP) ? "Stop" : "Start",
+             (hotkey_type == dsn::apps::hotkey_type::WRITE) ? "write" : "read",
+             app_name,
+             partition_index);
+    dsn::rpc_address meta_server;
+    meta_server.assign_group("meta-servers");
+    std::vector<dsn::rpc_address> meta_servers;
+    replica_helper::load_meta_servers(meta_servers);
+    for (const auto &address : meta_servers) {
+        meta_server.group_address()->add(address);
+    }
+    auto cluster_name = dsn::replication::get_current_cluster_name();
+    auto resolver = partition_resolver::get_resolver(cluster_name, meta_servers, app_name.c_str());
+    dsn::task_tracker tracker;
+    detect_hotkey_rpc rpc(
+        std::move(request), RPC_DETECT_HOTKEY, std::chrono::milliseconds(10), partition_index);
```

Comment:
```suggestion
        std::move(request), RPC_DETECT_HOTKEY, std::chrono::milliseconds(10), partition_index);
```

You really mean this RPC should respond within 10ms? Make it 10s.